src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -179,6 +183,41 @@ def sort_code_mapping_configs(
                     sorted_configs.insert(0, config)
         return sorted_configs
 
+    def get_codecov_path(self, source_url: str) -> Optional[str]:
+        # Extract path from URL from "blob/<sha_or_branch>/" onwards
+        blob_index = source_url.find("blob")
+        if blob_index != -1:
+            codecov_path_index = source_url.find("/", blob_index + len("blob/")) + len("/")
+            return source_url[codecov_path_index:]
+        return None
+
+    def get_codecov_line_coverage(
+        self, config: JSONData, source_url: str, access_token: str
+    ) -> Tuple[Optional[Sequence[Tuple[int, int]]], Optional[int]]:
+        owner_username, repo_name = config["repoName"].split("/")
+        url = CODECOV_URL.format(
+            service=config["provider"]["key"], owner_username=owner_username, repo_name=repo_name
+        )
+        path = self.get_codecov_path(source_url)
+        line_coverage, status_code = None, None
+        if path:
+            params = {"branch": config["defaultBranch"], "path": path}
+            try:
+                response = requests.get(
+                    url, params=params, headers={"Authorization": "tokenAuth %s" % access_token}
+                )
+                line_coverage, status_code = (
+                    response.json()["files"][0]["line_coverage"],
+                    response.status_code,
+                )
+            except Exception:
+                if status_code != 404:
+                    logger.exception("Failed to get coverage from Codecov")
+                line_coverage = None
```

Comment:
Does this need to be reset this here, given the assignment on L202?

Reply:
Good catch, thank you!